test/correctness/isnan.cpp

```diff
@@ -3,20 +3,14 @@
 
 using namespace Halide;
 
-int main(int argc, char **argv) {
-    Func f;
-    Var x;
-    Var y;
-
-    Expr e = sqrt(x - y);
-    f(x, y) = select(is_nan(e), 0.0f, 1.0f);
-    f.vectorize(x, 8);
+/* Halide uses fast-math by default. is_nan must either be used inside
+ * strict_float or as a test on inputs produced outside of
+ * Halide. Using it to test results produced by math inside Halide but
+ * not using strict_float is unreliable. This test covers both of these cases. */
```

Comment:
We can't immediately change the behavior of Halide::is_nan() in non-strict mode, but IMHO we should plan to deprecate it: perhaps if Halide::is_nan() is used in a non-strict Expr, we should emit a Warning now, with plans to convert that to a compile-time error in the future.

Reply:
(Related: we can now consider adding `Halide::is_finite()`, with the caveat that it errors when used in non-strict Exprs.)